sdb/commands/linux/whatis.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright 2020 Delphix
+# Copyright 2021 Datto Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import drgn.helpers.linux.mm as mm
+import sdb
+
+
+def lookup_in_kmem(obj: drgn.Object) -> None:
+    pfn = mm.virt_to_pfn(sdb.get_prog(), obj)
+    page = mm.pfn_to_page(pfn)
+    try:
+        cache = page.slab_cache
+        cache_nm = cache.name.string_().decode('utf-8')
+        print(f"{hex(int(obj))} is allocated from {cache_nm}")
+    except drgn.FaultError:
+        print(f"{hex(int(obj))} does not map to a kmem_cache")
```

Comment:
I think we should do a couple of changes to this helper if we are to truly make it re-usable:
[1] Move it to `sdb/commands/linux/internal/slub_helpers.py`
[2] substitute the `mm` alias with the conventions of the above helper file (e.g. import `virt_to_pfn` and `pfn_to_page` at the top of that file and use them without `mm`).
[3] Change the functions signature:
```
def lookup_cache_by_address(obj: drgn.Object) -> Optional[drgn.Object]:
```
[4] Move the printing to the sdb command itself and have that helper method return the cache as a drgn.Object. If we get a `FaultError` because the page is not part of the cache we return `None` and the caller/command needs to handle it.